tests/test_files.toml

```diff
@@ -537,3 +537,63 @@ code = '''
     }
     lfs_unmount(&lfs) => 0;
 '''
+
+[cases.test_files_ishandleopen]
+code = '''
+    lfs_t lfs;
+    lfs_format(&lfs, cfg) => 0;
+    lfs_mount(&lfs, cfg) => 0;
+    lfs_file_t file;
+
+    lfs_file_ishandleopen(&lfs, &file) => LFS_ERR_BADF;
+
+    lfs_file_open(&lfs, &file, "hello", LFS_O_CREAT | LFS_O_WRONLY) => 0;
+
+    lfs_file_ishandleopen(&lfs, &file) => 0;
+
+    lfs_file_close(&lfs, &file) => 0;
+
+    lfs_file_ishandleopen(&lfs, &file) => LFS_ERR_BADF;
```

Comment:
I hadn't thought about the return value of `lfs_file_ishandleopen`, LFS_ERR_BADF is a good choice. It's also the first use of this error code in the codebase.

Humorously, LFS_ERR_BADF is removed in v3-alpha, but we can bring it back for this purpose.

LFS_ERR_BADF also makes sense for a theoretical `LFS_UNTRUSTED_USER` build, where file operations error if the file is not open. Some users have asked for such a feature, but it's been low priority. (I'm also hesitant because it isn't bulletproof, we can't provide the same guardrails around if a filesystem is mounted, for example.)